lib/ReportConfigProcessor.php

```diff
@@ -0,0 +1,63 @@
+<?php
+namespace Octri\ConsortReport;
+
+require_once(dirname(realpath(__FILE__)) . '/ReportStrategy.php');
+
+/**
+ * A class for processing report configuration. Handles updating report summary
+ * configuration for the defined strategy.
+ */
+class ReportConfigProcessor {
+
+    private $report;
+    private $summaryConfig;
+
+    /**
+     * @param Array report The report data.
+     * @param Array summaryConfig The configuration for the report data. Used to
+     * build the <code>ReportSummary</code>.
+     */
+    public function __construct($report, $summaryConfig) {
+        $this->report = $report;
+        $this->summaryConfig = $summaryConfig;
+    }
+
+    /**
+     * Returns a flattened array containing only the values for the
+     * <code>bucketBy</code> field.
+     * @param String bucketBy The field name to bucket data by.
+     * @param Array data The report data.
+     * @return Array of values for the field bucketBy
+     */
+    private function mapBucketData() {
+        return array_map(
+            function($record, $bucketBy) {
+                return $record[$bucketBy];
+            },
+            $this->report,
+            array_fill(0, $this->summaryConfig['totalRecords'], $this->summaryConfig['bucket-by'])
```

Comment:
I'm not particularly happy with this mapping, but I wasn't sure of a better way. Improvements welcomed. What this is doing is taking `$this->report` which is of the form,

```
[
  [ "id": 1, "dsp_stop_reason": "foo" ],
  [ "id": 2, "dsp_stop_reason": "doo" ],
  ...
]
```

and mapping the value of `dsp_stop_reason`. So `mapBucketData` returns an array of those `dsp_stop_reason` values.

`array_map` allows for multiple arrays as inputs. The second array that I'm passing is an array where every value is `dsp_stop_reason`. Both arrays need to be of the same size. I need the `bucket-by` value in the anonymous function, but I couldn't find another way to inject the value.

Reply:
This might be a case where functional programming obscures what you're trying to do. Have you tried this as a `foreach`?

Reply:
A `foreach` would probably be a good choice. I looked a bit more and figured a simpler way to map this. I forgot about `use` on functions and I think it's pretty clear now. Something like,

```
$bucketBy = 'someField';
array_map(function($record) use ($bucketBy) {
    return $record[$bucketBy];
}, $this->report);
```